### PR TITLE
Added Environment Variables for setting Names of Resources

### LINKED ISSUE #1288: Use existing App Service and App Service Plan
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Mention any other details that might be useful
Hi, is it possible to use an existing App Service and App Service Plan from a previous deployment. Mainly to use the same URL as before. The same way we can use existing resources such as Azure OpenAI, AI Search, Document Intelligence and so on?
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by giterinhub**
Fixes #1288 

Added Environment Variables to allow setting the names of most resources.
+ App Service Plan Name
+ App Service Name (backend service)
+ Application Insights Name
+ Application Insights Dashboard Name
+ Log Analytics Workspace Name
+ Computer Vision Service Name
+ Computer Vision Resource Group Name
+ Computer Vision Location
+ Computer Vision SKU Name

Two Global Resources are created, whose names have **not** been parameterised. These are
- Action group (Application Insights Smart Detection)
- Smart detector alert rule (Failure Anomalies - $AZURE_APPLICATION_INSIGHTS)

The Environment Variables were also added to the Azure Pipeline and Github Action files:
* azure.yaml 
* .azdo/pipelines/azure-dev.yml
* .github/workflows/azure-dev.yml

## Purpose

To use a custom naming convention, or to be consistent in providing names to resources, additional Environment Variables are added for specifying the resource names that were not yet in the Parameters file.

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [X] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [X] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [ ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Comment by giterinhub** [comment]
@microsoft-github-policy-service agree

**Comment by giterinhub** [comment]
Hi @pamelafox - do you have any feedback for me about this PR?

**Comment by pamelafox** [comment]
I think it makes sense to allow more parameterization, thanks for the change! Can you add them to the workflow/pipeline file in this PR? I'm trying to remember to make sure we do that in any PR that adds env vars. (I forgot it in my embedding one but will as it soon)

**Review by pamelafox — CHANGES_REQUESTED** [review]
See my comment- please add to pipelines.

**Comment by giterinhub** [comment]
Ok, I updated workflow/azure-dev.yml but noticed there are still refs to AZURE_FORMRECOGNIZER_SERVICE where now it's called document intelligence? Anyway, should I update any other files? Thanks @pamelafox  and have a nice day!

**Comment by pamelafox** [comment]
@giterinhub Ah can you add the new DOC_INTEL variables? I suspected we missed some.
It'd be great if you could also add to the deploy_existing.md doc, with appropriate sections for each service.

**Comment by giterinhub** [comment]
> @giterinhub Ah can you add the new DOC_INTEL variables? I suspected we missed some. It'd be great if you could also add to the deploy_existing.md doc, with appropriate sections for each service.

Hi @pamelafox thank you for your feedback, I've applied all the changes, and the tests all passed. Please review.

File: infra/main.parameters.json [hunk 1]
```diff
@@ -23,6 +23,18 @@
...
       "value": "S0"
     },
+    "computerVisionServiceName": {
+      "value": "${AZURE_COMPUTER_VISION_SERVICE}"
+    },
+    "computerVisionResourceGroupName": {
+      "value": "${AZURE_COMPUTER_VISION_RESOURCE_GROUP}"
+    },
+    "computerVisionResourceGroupLocation": {
+      "value": "${AZURE_COMPUTER_VISION_LOCATION}"
```
Comment (pamelafox) [inline infra/main.parameters.json:33]: I think this one needs to default to eastus, as thats what main.bicep has currently, and then main.bicep should be set to nothing with a comment "// Set in main.parameters.json"

File: infra/main.parameters.json [hunk 2]
```diff
@@ -23,6 +23,18 @@
...
+    },
+    "computerVisionSkuName": {
+      "value": "${AZURE_COMPUTER_VISION_SKU=F0}"
```
Comment (pamelafox) [inline infra/main.parameters.json:36]: This has the wrong default value, it should be S1

File: infra/main.parameters.json [hunk 3]
```diff
@@ -68,9 +80,15 @@
     "storageSkuName": {
       "value": "${AZURE_STORAGE_SKU=Standard_LRS}"
     },
+    "appServicePlanName": {
```
Comment (pamelafox) [inline infra/main.parameters.json:83]: For all these who have an empty string value in main.parameters.json, please just remove that empty string and put a comment of "// Set in main.parameters.json"

**Review by pamelafox — CHANGES_REQUESTED** [review]
Thanks! We need some changes to make this change compatible with environments that don't have these values set.

File: infra/main.parameters.json (hunk 1 above)
Comment (giterinhub) [reply infra/main.parameters.json:33]: AI Vision and Document Intelligence is now widely available, but I'll set the default as you suggest. 
https://azure.microsoft.com/en-gb/explore/global-infrastructure/products-by-region/?products=cognitive-services

File: infra/main.parameters.json (hunk 2 above)
Comment (giterinhub) [reply infra/main.parameters.json:36]: Default value in param updated to S1, comment added in main.bicep.

File: infra/main.parameters.json (hunk 3 above)
Comment (giterinhub) [reply infra/main.parameters.json:83]: Updated all the relevant variables.

**Review by pamelafox — APPROVED** [review]
I've done several manual tests, this looks good now. I realized I had to set params to '' in main.bicep to avoid being prompted for the values.

**Comment by pamelafox** [comment]
Thank you!

**Pull request merged**